src/OrbitGl/App.cpp

```diff
@@ -1823,16 +1831,20 @@ orbit_base::Future<ErrorMessageOr<std::filesystem::path>> OrbitApp::RetrieveModu
     return local_symbols_future;
   }
 
+  orbit_base::StopSource stop_source{};
+
   Future<ErrorMessageOr<std::filesystem::path>> final_result =
       orbit_base::UnwrapFuture(local_symbols_future.Then(
           main_thread_executor_,
-          [this, module_id](ErrorMessageOr<std::filesystem::path> local_symbols_path)
+          [this, module_id, stop_token = stop_source.GetStopToken()](
+              ErrorMessageOr<std::filesystem::path> local_symbols_path)
               -> Future<ErrorMessageOr<std::filesystem::path>> {
             if (local_symbols_path.has_value()) {
+              modules_currently_loading_.erase(module_id);
```

Comment:
Why was that not needed before? Also is it save to read/write into this map from different threads?

Reply:
It was forgotten before and it basically didn't have any downside. I am going to use the list `modules_currently_loading_` in an upcoming PR to determine whether a module is currently downloaded and display that in the ModuleList. If this erase here is missing, the module list will always display "Downloading...".

Reply:
Regarding thread safety: This is not safe to be accessed from any other thread than the main thread. This is currently not enforced. I added a comment and this will be tackled as part of a bigger symbol loading refactoring. This will move to a separate class